src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/SimdVectorExtensions.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.Intrinsics
+{
+    internal static class SimdVectorExtensions
+    {
+        /// <summary>Copies a vector to a given array.</summary>
+        /// <typeparam name="TVector">The type of the vector.</typeparam>
+        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
+        /// <param name="vector">The vector to be copied.</param>
+        /// <param name="destination">The array to which <paramref name="vector" /> is copied.</param>
+        /// <exception cref="ArgumentException">The length of <paramref name="destination" /> is less than <see cref="ISimdVector{TVector, T}.Count" />.</exception>
+        /// <exception cref="NullReferenceException"><paramref name="destination" /> is <c>null</c>.</exception>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        public static void CopyTo<TVector, T>(this TVector vector, T[] destination)
+            where TVector : unmanaged, ISimdVector<TVector, T>
+            where T : struct
```

Comment:
This is being "clever" to continue allowing these as extension methods for perf reasons (instance methods on value types take the first parameter as a hidden byref, which is undesirable for types that should always be enregistered and almost never address taken).